tests/common/plugins/pdu_controller/pdu_manager.py

```diff
@@ -163,10 +172,10 @@ def get_outlet_status(self, outlet=None):
             status = status + outlets
         else:
             # collect all status
-            for pdu_index, controller in enumerate(self.controllers):
-                if len(controller['outlets']) > 0:
-                    outlets = controller['controller'].get_outlet_status(hostname=self.dut_hostname)
-                    self._update_outlets(outlets, pdu_index)
+            for controller_index, controller in enumerate(self.controllers):
+                for outlet in controller['outlets']:
+                    outlets = controller['controller'].get_outlet_status(outlet=outlet['outlet_id'])
+                    self._update_outlets(outlets, outlet['pdu_index'], controller_index)
```

Comment:
for the case of 2 outlets on the same PDU, will this loop always updating the first outlet in record? because both outlet will have same 'pdu_index' 0, and since we query them one at a time, they will both be outlet_index 0?

Reply:
The outlet's PDU index is 0 and 1, not both 0. In the add_controller(), there is a next_index to calculate the pdu_index accordingly.